server/neptune/workflows/internal/activities/deployment/store.go

```diff
@@ -0,0 +1,132 @@
+package deployment
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/graymeta/stow"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deployment"
+	"github.com/uber-go/tally/v4"
+)
+
+const OutputPrefix = "deployments"
+
+func NewStore(deploymentCfg valid.Deployments, scope tally.Scope) (*instrumentedStore, error) {
+	if deploymentCfg.StorageBackend == nil {
+		return nil, errors.New("error initializing deployment info store")
+	}
+
+	config := deploymentCfg.StorageBackend.BackendConfig.GetConfigMap()
+	backend := deploymentCfg.StorageBackend.BackendConfig.GetConfiguredBackend()
+	containerName := deploymentCfg.StorageBackend.BackendConfig.GetContainerName()
+
+	location, err := stow.Dial(backend, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &instrumentedStore{
+		store: &store{
+			location:      location,
+			containerName: containerName,
+		},
+		scope: scope.SubScope("store"),
+	}, nil
+
+}
+
+type store struct {
+	location      stow.Location
+	containerName string
+}
+
+func (s *store) GetDeploymentInfo(ctx context.Context, repoName string, rootName string) (*deployment.Info, error) {
+	container, err := s.location.Container(s.containerName)
+	if err != nil {
+		return nil, errors.Wrap(err, "resolving container")
+	}
+
+	key := buildKey(repoName, rootName)
+	item, err := container.Item(key)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting item")
+	}
+
+	r, err := item.Open()
+	if err != nil {
+		return nil, errors.Wrap(err, "reading item")
+	}
+
+	decoder := json.NewDecoder(r)
```

Comment:
This happens in multiple places like the job store as well, is there anyway we can dedup this code and make some client that can be shared? It seems weird that other places in the code base need to know how to interact with stow low level constructs.

Reply:
True! I created a `stow.Client` that deals with stow specific stuff and used that client in deployments store and jobs store